src/main/app/src/store/reducers/toteutusSlice.js

```diff
@@ -71,19 +73,82 @@ const isHakuEndInFuture = (hakuajat) => {
   return hakuajat.some((aika) => !aika.paattyy || isBefore(now, new Date(aika.paattyy)));
 };
 
-const selectHakukohteet = (state, oid, type) =>
-  state.toteutus.toteutukset[oid]?.hakutiedot
+const getHakukohteetWithType = (toteutus, type) =>
+  toteutus.hakutiedot
     ?.filter((hakutieto) => hakutieto.hakutapa.nimi.fi === type)
     .map((hakutieto) => hakutieto.hakukohteet)
     .flat()
     .filter((hakukohde) => isHakuEndInFuture(hakukohde.hakuajat))
     .map((hakukohde) => ({ ...hakukohde, isHakuAuki: isHakuAuki(hakukohde.hakuajat) }));
 
 export const selectLoading = (state) => state.toteutus.status === LOADING_STATUS;
-export const selectJatkuvatHaut = (oid) => (state) =>
-  selectHakukohteet(state, oid, JATKUVAHAKU);
-export const selectErillisHaut = (oid) => (state) =>
-  selectHakukohteet(state, oid, ERILLISHAKU);
-export const selectYhteisHaut = (oid) => (state) =>
-  selectHakukohteet(state, oid, YHTEISHAKU);
-export const selectToteutus = (oid) => (state) => state.toteutus.toteutukset[oid];
+export const selectHakukohteet = (oid) => (state) => {
+  const toteutus = selectToteutus(oid)(state);
+  if (!toteutus || toteutus.hasMuuHaku || toteutus.hasEiSahkoistaHaku) {
+    return {};
+  }
+
+  return {
+    jatkuvatHaut: getHakukohteetWithType(toteutus, JATKUVAHAKU),
+    erillisHaut: getHakukohteetWithType(toteutus, ERILLISHAKU),
+    yhteisHaut: getHakukohteetWithType(toteutus, YHTEISHAKU),
+  };
+};
+
+export const selectMuuHaku = (oid) => (state) => {
+  const toteutus = selectToteutus(oid)(state);
+  const hakuAuki = isHakuAuki([toteutus.metadata.hakuaika]);
+
+  // TODO: SORA-kuvaus - atm. we only get an Id from the API but we cannot do anything with it
+  return {
+    ...pick(toteutus.metadata, [
+      'aloituspaikat',
+      'hakuaika',
+      'hakulomakeLinkki',
+      'hakutermi',
+      'lisatietoaHakeutumisesta',
+      'lisatietoaValintaperusteista',
+    ]),
+    isHakuAuki: hakuAuki,
+    nimi: toteutus.nimi,
+    // TODO: we do not get osoite from the API atm. so just use all the tarjoajat to fetch oppilaitoksenOsat
+    // This should be replaced with just the osoite when we have it in the API
+    tarjoajat: toteutus.tarjoajat,
+  };
+};
+
+export const selectEiSahkoistaHaku = (oid) => (state) => {
+  const toteutus = selectToteutus(oid)(state);
+  return {
+    ...pick(toteutus.metadata, ['lisatietoaHakeutumisesta']),
+  };
+};
+
+const isAnyHakuAuki = (toteutus) => {
```

Comment:
Funktion nimi antaa tässä olettaa, että palauttaisi booleanin, mutta näin ei ole.